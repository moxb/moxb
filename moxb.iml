<?xml version="1.0" encoding="UTF-8"?>
<module type="WEB_MODULE" version="4">
  <component name="NewModuleRootManager" inherit-compiler-output="true">
    <exclude-output />
    <content url="file://$MODULE_DIR$">
      <sourceFolder url="file://$MODULE_DIR$/packages/antd/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/meteor/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/moxb/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/packages/react-html/src" isTestSource="false" />
      <sourceFolder url="file://$MODULE_DIR$/examples/src" isTestSource="false" />
      <excludeFolder url="file://$MODULE_DIR$/packages/moxb/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/meteor/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/react-html/dist" />
      <excludeFolder url="file://$MODULE_DIR$/packages/antd/dist" />
      <excludeFolder url="file://$MODULE_DIR$/admin/node-installation" />
      <excludeFolder url="file://$MODULE_DIR$/admin/bin-tools" />
      <excludeFolder url="file://$MODULE_DIR$/examples/dist" />
      <excludeFolder url="file://$MODULE_DIR$/examples/.cache" />
      <excludeFolder url="file://$MODULE_DIR$/coverage" />
    </content>
    <orderEntry type="inheritedJdk" />
    <orderEntry type="sourceFolder" forTests="false" />
  </component>
</module>