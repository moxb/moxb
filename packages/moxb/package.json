{
  "name": "@moxb/moxb",
  "version": "0.2.0-beta.59",
  "description": "MobX data binding",
  "scripts": {
    "test": "echo \"Error: no test specified\" && exit 1"
  },
  "main": "dist",
  "typings": "dist/",
  "files": [
    "dist/**/*"
  ],
  "repository": {
    "type": "git",
    "url": "https://github.com/moxb/moxb.git"
  },
  "author": "moxb (moxb.org)",
  "license": "MIT",
  "bugs": {
    "url": "https://github.com/moxb/moxb/issues"
  },
  "homepage": "https://github.com/moxb/moxb#readme",
  "keywords": [
    "mobx",
    "UI"
  ],
  "jest": {
    "moduleFileExtensions": [
      "ts",
      "tsx",
      "js",
      "jsx",
      "json",
      "node"
    ],
    "transform": {
      "^.+\\.(ts|tsx)$": "ts-jest"
    },
    "testEnvironment": "node",
    "testMatch": [
      "**/src/**/__tests__/*.(ts|tsx)"
    ],
    "modulePaths": [
      "<rootDir>/peer-dependencies/node_modules"
    ]
  },
  "peerDependencies": {
    "mobx": "^5.15.6",
    "tslib": "^2.0.1"
  },
  "devDependencies": {
    "@types/atob": "2.1.2",
    "@types/jest": "26.0.13",
    "@types/lodash.flatten": "4.4.6",
    "@types/urijs": "1.19.11",
    "mobx": "5.13.1",
    "tslib": "1.10.0"
  },
  "dependencies": {
    "btoa-lite": "1.0.0",
    "history": "5.0.0",
    "lodash.flatten": "4.4.0",
    "moment": "2.29.1",
    "urijs": "1.19.2"
  }
}
