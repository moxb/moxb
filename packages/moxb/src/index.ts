export * from './impl';
export * from './types';
export * from './routing';
