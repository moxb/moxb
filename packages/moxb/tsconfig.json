{
    "compilerOptions": {
        "module": "commonjs",
        "target": "ES5",
        "jsx": "react",
        "lib": ["dom","es6","es2015","es2017", "dom.iterable","scripthost"],
        "inlineSourceMap": true,
        "importHelpers":true,
        "experimentalDecorators": true,
        "noImplicitAny": true,
        "noImplicitThis": true,
        "strict": true,
        "outDir": "./dist",
        "declaration": true,
        "skipLibCheck": true,
        "strictNullChecks": true,
        "removeComments": false,
        "noUnusedLocals": true,
        "noUnusedParameters": true,
        "allowUnreachableCode": false,
        "baseUrl": "./src",
        "paths": {
            "*": [
                "*",
                "../../../node_modules/*"
            ]
        }
    },
    "exclude": [
        "./node_modules",
        "./dist"
    ]
}
