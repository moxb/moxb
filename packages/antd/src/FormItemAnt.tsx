import { FormItem as MoxFormItem } from '@moxb/moxb';
import { Form } from 'antd';
import { FormItemProps } from 'antd/lib/form/FormItem';
import { ColProps } from 'antd/lib/grid';
import { observer } from 'mobx-react';
import * as React from 'react';
import { CSSProperties } from 'react';
import { getErrorMessages, labelWithHelpSpan, parseProps } from './BindAnt';
import { BindMarkdownDiv } from './LabelAnt';

// Due to the recent antd changes the children property in the FormItemProps component
// became mandatory which is invalid
export interface BindFormItemAntProps extends Partial<FormItemProps> {
    operation: MoxFormItem;
    formStyle?: CSSProperties;
    wrapperCol?: ColProps;
    labelCol?: ColProps;
    label?: string;
    extraMarkdown?: string;
}

@observer
export class FormItemAnt extends React.Component<BindFormItemAntProps> {
    render() {
        const {
            operation,
            id,
            children,
            label,
            help,
            formStyle,
            labelCol,
            wrapperCol,
            invisible,
            extra,
            extraMarkdown,
        } = parseProps(this.props, this.props.operation);
        if (invisible) {
            return null;
        }
        const extraLabel = extraMarkdown && !extra ? <BindMarkdownDiv text={extraMarkdown} /> : extra;
        const fullId = id + '-formItem';
        return (
            <Form.Item
                htmlFor={fullId}
                data-testid={fullId}
                label={labelWithHelpSpan(label, help)}
                style={formStyle || undefined}
                extra={extraLabel}
                labelCol={labelCol}
                wrapperCol={wrapperCol}
                required={operation.required}
                hasFeedback={operation.hasErrors}
                validateStatus={operation.hasErrors ? 'error' : undefined}
                help={operation.hasErrors ? getErrorMessages(operation.errors!) : null}
            >
                {children}
            </Form.Item>
        );
    }
}

/**
 * This function filters out the props, which are not needed by a FormItem children.
 * It only passes the necessary props along to the children.
 */
export function parsePropsForChild<T, O>(bindProps: T, op: O): T & O {
    // @ts-ignore
    const { label, formStyle, labelCol, wrapperCol, ...props } = parseProps(bindProps, op);
    return props as T & O;
}
