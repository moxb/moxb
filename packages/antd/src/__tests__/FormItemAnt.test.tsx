import { FormImpl, TextImpl } from '@moxb/moxb';
import { shallow } from 'enzyme';
import * as React from 'react';
import { FormItemAnt } from '../FormItemAnt';

describe('FormItemAnt', function() {
    const formUserText = new TextImpl({
        id: 'ApplicationImpl.formUserText',
        label: 'Username',
        onSave: (_bind, _done) => jest.fn(),
    });
    const formPasswordText = new TextImpl({
        id: 'ApplicationImpl.formUserText',
        label: 'Username',
        onSave: (_bind, _done) => jest.fn(),
    });

    it('should render a form item by default', function() {
        const onSubmitMock = jest.fn();
        const operation = new FormImpl({
            id: 'id.testForm',
            values: [formUserText, formPasswordText],
            onSubmit: (_bind, _done) => onSubmitMock,
        });
        expect(shallow(<FormItemAnt operation={operation} />)).toMatchSnapshot();
    });

    it('should return null if invisible', function() {
        const onSubmitMock = jest.fn();
        const operation = new FormImpl({
            id: 'id.testForm',
            values: [formUserText, formPasswordText],
            onSubmit: (_bind, _done) => onSubmitMock,
            invisible: () => true,
        });
        expect(shallow(<FormItemAnt operation={operation} />).type()).toBeNull();
    });

    it('should show an error message on failure', function() {
        const onSubmitMock = jest.fn();
        const operation = new FormImpl({
            id: 'id.testForm',
            values: [formUserText, formPasswordText],
            onSubmit: (_bind, _done) => onSubmitMock,
        });
        const wrapper = shallow(<FormItemAnt operation={operation} />);
        formUserText.setError('New error');
        expect(wrapper.render()).toMatchSnapshot();
        expect(wrapper.render().find('div.ant-form-explain').length).toBe(1);
    });
});
