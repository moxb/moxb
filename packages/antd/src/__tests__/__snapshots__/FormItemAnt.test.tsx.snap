// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FormItemAnt should render a form item by default 1`] = `
<FormItem
  data-testid="id-test_form-formItem"
  hasFeedback={false}
  help={null}
  htmlFor="id-test_form-formItem"
/>
`;
